mobile/src/main.jsx

```diff
@@ -10,102 +10,88 @@ import { Provider } from 'react-redux'
-
 const loggerMiddleware = createLogger()
 
-document.addEventListener('DOMContentLoaded', () => {
-  loadState().then(persistedState => {
-    const root = document.querySelector('[role=application]')
+const onError = (store, callback) => () => {
+  console.warn(`Your device is no more connected to your server: ${store.getState().mobile.settings.serverUrl}`)
+  store.dispatch(revokeClient())
+  callback()
+}
 
-    const store = createStore(
-      filesApp,
-      persistedState,
-      applyMiddleware(
-        RavenMiddleWare(`https://${__SENTRY_TOKEN__}@sentry.cozycloud.cc/2`),
-        thunkMiddleware,
-        loggerMiddleware
-      )
+const renderAppWithPersistedState = persistedState => {
+  const store = createStore(
+    filesApp,
+    persistedState,
+    applyMiddleware(
+      RavenMiddleWare(`https://${__SENTRY_TOKEN__}@sentry.cozycloud.cc/2`),
+      thunkMiddleware,
+      loggerMiddleware
     )
+  )
 
-    store.subscribe(() => {
-      const stateToBeSaved = {
-        mobile: {
-          settings: store.getState().mobile.settings,
-          mediaBackup: {
-            uploaded: store.getState().mobile.mediaBackup.uploaded
-          }
-        }
+  store.subscribe(() => saveState({
+    mobile: {
+      settings: console.log(store.getState().mobile.settings) || store.getState().mobile.settings,
+      mediaBackup: {
+        uploaded: store.getState().mobile.mediaBackup.uploaded
       }
-      saveState(stateToBeSaved)
-    })
+    }
+  }))
 
-    initClient(store.getState().mobile.settings.serverUrl)
+  initClient(store.getState().mobile.settings.serverUrl)
 
-    function requireSetup (nextState, replace, callback) {
-      const client = store.getState().mobile.settings.client
-      const isSetup = store.getState().mobile.settings.authorized
-      if (isSetup) {
-        isClientRegistered(client).then(clientIsRegistered => {
-          if (clientIsRegistered) {
-            const options = {
-              onError: (err) => {
-                console.log('on error fron the client', err)
-                console.warn(`Your device is no more connected to your server: ${store.getState().mobile.settings.serverUrl}`)
-                store.dispatch(revokeClient())
-              }
-            }
-            cozy.client.offline.startRepeatedReplication('io.cozy.files', 15, options)
-            initBar()
-            callback()
-          } else {
-            console.warn(`Your device is no more connected to your server: ${store.getState().mobile.settings.serverUrl}`)
-            store.dispatch(revokeClient())
-            callback()
+  function requireSetup (nextState, replace, callback) {
+    const client = store.getState().mobile.settings.client
+    const isSetup = store.getState().mobile.settings.authorized
+    if (isSetup) {
+      isClientRegistered(client).then(clientIsRegistered => {
+        if (clientIsRegistered) {
+          const options = {
+            onError: onError(store, callback)
```

Comment:
We could add `err` parameter in the onError HOF here https://github.com/cozy/cozy-files-v3/pull/154/files/566f31ecc6325cd4dd01b1d32d9322beb1b67a71#diff-c609607c962d635b37d6c6f38848804fR27

But with standardjs, we don't want to have an error not handled: `cozy-files-v3/mobile/src/main.jsx:27:38: Expected error to be handled.`. And because we have nothing to handle... I do not add the err parameter.